cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go

```diff
@@ -23,252 +23,271 @@ limitations under the License.
-// for kube proxy official documentation.
-//
-//    apiVersion: kubelet.config.k8s.io/v1beta1
-//    kind: KubeletConfiguration
-//       ...
-//
-// The KubeletConfiguration type should be used to change the configurations that will be passed to all kubelet instances
+// **Kubeadm init configuration types**
+//
+// When executing kubeadm init with the `--config` option, the following configuration types could be used:
+// `InitConfiguration`, `ClusterConfiguration`, `KubeProxyConfiguration`, `KubeletConfiguration`, but only one
+// between `InitConfiguration` and `ClusterConfiguration` is mandatory.
+//
+// ```yaml
+// apiVersion: kubeadm.k8s.io/v1beta2
+// kind: InitConfiguration
+// bootstrapTokens:
+//   # ...
+// nodeRegistration:
+//   # ...
+// ```
+//
+// The `InitConfiguration` type is used to configure runtime settings. In the case of `kubeadm init`,
+// it includes the configuration of the bootstrap token and all the setting specific to the node
+// where kubeadm is executed, including:
+//
+// - `nodeRegistration`: fields that relate to registering the new node to the cluster.
+//   You can use it to customize the node name, the CRI socket to use or any other
+//   settings that should apply to this node only (for example. the node IP).
+//
+// - `localAPIEndpoint`: the endpoint of the API server instance to be deployed on this node.
+//   For example, you can use it to customize the API server advertise address.
+//
+//   ```yaml
+//   apiVersion: kubeadm.k8s.io/v1beta2
+//   kind: ClusterConfiguration
+//   networking:
+//     # ...
+//   etcd:
+//     # ...
+//   apiServer:
+//     extraArgs:
+//       # ...
+//     extraVolumes:
+//       # ...
+//   # ...
+//   ```
+//
+// The `ClusterConfiguration` type can be used to configure cluster-wide settings, including:
+//
+// - Networking, that holds configuration for the networking topology of the cluster;
+//   For example, uou can use it to customize node subnet or services subnet.
+//
+// - Etcd configurations that can be used for customizing the local etcd or to configure
+//   the API server for using an external etcd cluster.
+//
+// - kube-apiserver, kube-scheduler, kube-controller-manager configurations.
+//   You can use it to customize control-plane components by adding customized setting
+//   or overriding kubeadm default settings.
+//
+// ```yaml
+// apiVersion: kubeproxy.config.k8s.io/v1alpha1
+// kind: KubeProxyConfiguration
+// # ...
+// ```
+//
+// The `KubeProxyConfiguration` type is used to change the configurations passed to the kube-proxy
+// instances deployed in the cluster. If this object is not provided or provided only partially,
+// kubeadm applies defaults.
+// See [kube-proxy reference](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/)
+// or [kube-proxy source code](https://godoc.org/k8s.io/kube-proxy/config/v1alpha1#KubeProxyConfiguration)
```

Comment:
"kube-proxy API documentation"?

Reply:
alternatively we can call these links (here and in other places):

`[kube-proxy command line reference]` and `[kube-proxy API reference]`